src/main/java/com/cloudbees/jenkins/support/util/WordReplacer.java

```diff
@@ -161,11 +169,14 @@ private static void replaceWord(StringBuilder input, String word, String replace
                 }
             }
 
-            // replace this word but in the original input (without changing the case)
-            input.replace(pos, pos + workWord.length(), replace);
+            if (pos >= 0 && pos < input.length()) {
```

Comment:
Probably out of scope of this PR, but I'm starting to think it would be interesting to have some utility functions to improve the readability here.

Reply:
It could happen if the replacement word has less length